src/core/Akka.Cluster/Gossip.cs

```diff
@@ -442,43 +442,40 @@ public GossipOverview Copy(ImmutableHashSet<UniqueAddress> seen = null, Reachabi
     /// the node with same host:port. The `uid` in the `UniqueAddress` is
     /// different in that case.
     /// </summary>
-    class GossipEnvelope : IClusterMessage
+    internal class GossipEnvelope : IClusterMessage
     {
-        readonly UniqueAddress _from;
-        readonly UniqueAddress _to;
-
-        /// <summary>
-        /// TBD
-        /// </summary>
-        /// <param name="from">TBD</param>
-        /// <param name="to">TBD</param>
-        /// <param name="gossip">TBD</param>
-        /// <param name="deadline">TBD</param>
-        /// <returns>TBD</returns>
         public GossipEnvelope(UniqueAddress from, UniqueAddress to, Gossip gossip, Deadline deadline = null)
         {
-            _from = from;
-            _to = to;
+            From = from;
+            To = to;
             Gossip = gossip;
             Deadline = deadline;
         }
 
         /// <summary>
-        /// TBD
+        /// The sender of the gossip.
         /// </summary>
-        public UniqueAddress From { get { return _from; } }
+        public UniqueAddress From { get; }
+
         /// <summary>
-        /// TBD
+        /// The receiver of the gossip.
         /// </summary>
-        public UniqueAddress To { get { return _to; } }
+        public UniqueAddress To { get; }
+
         /// <summary>
-        /// TBD
+        /// The gossip content itself
         /// </summary>
-        public Gossip Gossip { get; set; }
+        public Gossip Gossip { get; }
+        
         /// <summary>
-        /// TBD
+        /// The deadline for the gossip.
         /// </summary>
         public Deadline Deadline { get; set; }
+
+        public override string ToString()
+        {
+            return $"GossipEnvelope(from={From}, to={To}, gossip={Gossip}, deadline={Deadline})";
```

Comment:
This is the primary `DeadLetter` logging change I wanted to make here - should make it much easier to understand what was lost in the `DeadLetter` pile in terms of missing gossip messages when there's things shutting down inside cluster nodes.